### ISSUE TITLE
LE ssl not working for IDN domains

### ISSUE THREAD

**Issue by EdNett** [omega8cc/boa#968]
Thanks again for the fantastic Let's Encrypt. 

1) We can't get it to work on IDN's - is this something we need to ask / report elsewhere?

2) For a few regular domains we can't get it to work,  (although it is working fine for most domains) . Are there suppossed to be 2 control .pid files for LE ssl successfully enabled domains - for each domain in
/aegir root/tools/le/ctl/

a live-example.com.pid  as well as an example.com.pid  ?  - I ask this because on a vps that has both .pid files per domain, it is working perfectly. On a vps that only has the example.com.pid - the ssl is not using the LS SSL.  Disabling it, reverifying, clearing caches, and then enabling it again does not solve the problem. I'll wait a day and then redo the up-stable for both barracuda and octopus, and see if that solves the problem.

The domains do have example.com and www.example.com DNS setup properly and there is "no redirection" set in the site edit.

BOA info more is: https://gist.github.com/EdNett/95415a6b9ee8558fc6bef2666cf05a21

and error messages are here: https://gist.github.com/EdNett/57bb6f73e14b8005bec5bb1b824f1382

**Comment by omega8cc** [comment]
Let's Encrypt doesn't support IDN domains, yet: https://letsencrypt.org/upcoming-features/

**Comment by omega8cc** [comment]
As for the error:
```
[hosting_le] The LE Certificate attempted creation failed in /data/disk/fast1/tools/le/certs/example.com
[hosting_le] The LE Certificate attempted creation failed in /data/disk/fast1/tools/le/certs/example2.com	
```

It's weird. Have you tried to rename the site, perhaps?

Please try to disable SSL on the affected site and then enable it again.

**Comment by EdNett** [comment]
Hi,

For the non-IDN domains, I have resolved the problem on one vps for one site that wasn't working properly with LE - there was an example.com entry in
/data/disk/o1/config/ssl.d
but there was no example.com.bak entry

same for /data/disk/o1/config/server_master/ssl.d  -  no example.com.bak file

I guess that the verify task must have aborted for some reason when trying to make the .bak file.  In any case another verification of the example.com domain did the trick and created the missing .bak files.

I still have out test vps for which neither test site works with LE - the error message above states that
[hosting_le] The LE Certificate attempted creation failed in /data/disk/fast1/tools/le/certs/example.com

however they are being created. Every time I verify the site, it creates 3 more files:
a cert-xxx.csr
a cert-xxx.pem
and a 
privkey-xxx.pem 

I tried modifying the permissions of the three files to give the group read permissions and re-verify but that did not help.  Does this give you any ideas about what is happening here?

There are no invalid server names, and service nginx reload doesn't give any errors, just the following warning:
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found

Do you think it's important to troubleshoot this further? If not I could recreate the sites - they are test sites used only for test updating.

**Comment by omega8cc** [comment]
We couldn't reproduce this issue. No, you should never touch any files there. If something doesn't work, disable SSL on that site, rename it to some other valid name and then enable SSL again. Then check the task log for further hints, if needed.

**Comment by EdNett** [comment]
IDN Support with Let's Encrypt has been enabled as of Oct. 20 2016 - can't we get this working? (We'd love to help you test this, and hava current stable BOA's with IDn"s installed)
Enabled: October 20, 2016
Let’s Encrypt now supports issuance for Internationalized Domain Names (IDNs).

**Comment by EdNett** [comment]
This is now fixed for BOA 3.2.2 stable - IDN support in dehydrated/LE is working.
